src/views/docs-view/server.ts

```diff
@@ -275,49 +287,40 @@ export function getStaticGenerationFunctions<
 			/**
 			 * Constructs the base sidebar level for `DocsView`.
 			 */
+			const docsSidebarTitle =
+				currentRootDocsPath.shortName || currentRootDocsPath.name
+			const docsBasePathFullPath = versionPathPart
+				? `/${product.slug}/${basePath}/${versionPathPart}`
+				: `/${product.slug}/${basePath}`
 			const docsSidebarLevel: SidebarProps = {
 				backToLinkProps: getBackToLink(currentRootDocsPath, product),
 				levelButtonProps: {
 					levelUpButtonText: `${product.name} Home`,
 				},
 				menuItems: navDataWithFullPaths as EnrichedNavItem[],
-				title: currentRootDocsPath.shortName || currentRootDocsPath.name,
-			}
-			/**
-			 * In some cases, the first nav item is a heading.
-			 * In these case, we'll visually hide the sidebar title,
-			 * since it will redundant right next to the authored title.
-			 */
-			const firstItemIsHeading =
-				typeof navDataWithFullPaths[0]?.heading == 'string'
-			if (firstItemIsHeading) {
-				docsSidebarLevel.visuallyHideTitle = true
+				title: docsSidebarTitle,
+				/* We always visually hide the title, as we've added in a
+				   "highlight" item that would make showing the title redundant. */
+				visuallyHideTitle: true,
 			}
 
 			/**
-			 * Check the top level of the navData for "overview" items,
-			 * which are expected to be present for consistency.
-			 * If we do no have an overview item match, then we'll
-			 * automatically add an overview item.
-			 */
-			const overviewItemMatch = navDataWithFullPaths.find((item: MenuItem) => {
-				const isPathMatch =
-					item.path == '' ||
-					item.path == '/' ||
-					item.path == '/index' ||
-					item.path == 'index'
-				return isPathMatch
-			})
-			/**
-			 * Exception: If the first navData node is a `heading`,
-			 * we'll avoid adding an overview item even if there's
-			 * no overview item match.
+			 * We always include a brand-themed "highlight" item,
+			 * which serves as a menu item representing the top-level docs category.
+			 *
+			 * If the first item in docsSidebarLevel.menuItems is an "Overview" item,
+			 * then we remove it, as it would be redundant with the "highlight" item
+			 * we're adding.
 			 */
-			if (!overviewItemMatch && !firstItemIsHeading) {
-				docsSidebarLevel.overviewItemHref = versionPathPart
-					? `/${product.slug}/${basePath}/${versionPathPart}`
-					: `/${product.slug}/${basePath}`
+			if (isOverviewItem(docsSidebarLevel.menuItems[0])) {
```

Comment:
💭 Again, here I wonder why we can't edit the source files. I'm sure there's a bigger conversation about that I haven't seen.

Reply:
Probably because coordinating that effort would be difficult? I'd just be worried about the application taking on too many of these types of things, instead of updating source content, which would be cleaner /easier in the long run.

Reply:
This is a great question. For docs views, I think editing the source files could be an option. For other views we don't necessarily have the same author interface anyways, but I'm assuming this question was mainly focused on docs.

I think what made me lean towards implementing on our side is how restrictive I think we'd be with adding in "highlight" items. I think:
- We'd only ever want authors to include these "highlight" items as the first item in a sidebar
- We always want the "highlight" item's title to be the name of the docs category
- We always want the "highlight" item's URL to be the root of the docs category

If that all seems right, then I think it might be a little odd to expose highlight items as authorable in source - we'd want authors to use them in a very specific way, so they wouldn't exactly be "authorable" in that sense.

All that being said -- for the Packer plugins situation I think you're absolutely right, we should ideally update in source 😅 During QA there are also a few spots in Terraform content where we have similar situations.

💭 Of course, the biggest challenge as you mentioned is co-ordination.... I think I'll work on a clearer workflow for adding a "highlight" item to `menuItems` - it could handle the removal of "Overview" items for now, and ideally handle the Terraform and Packer plugin cases in a more clean and robust way. Once content is updated, then we can remove those extra processing step on our end. Hopefully this should address the Packer plugin case brittleness as well. Will update here if I get anywhere with that!

Reply:
Update: implemented a helper function `addBrandedOverviewSidebarItem` to do this. See [this comment](https://github.com/hashicorp/dev-portal/pull/1552#issuecomment-1404565590) for details.